### ISSUE TITLE
Should `Style/StringLiterals` also apply to quoted symbols?

### ISSUE THREAD

**Issue by sambostock** [rubocop/rubocop#9793]
- `Style/StringLiterals` provides the ability to enforce a consistent quoting style (`"..."` vs `'...'`), with rich support for edge cases.
- `Lint/SymbolConversion` provides the ability to enforce consistent symbol literal syntax (again with rich support for edge cases), and respects the style specified by `Style/StringLiterals` when autocorrecting by inserting quotes.

However, when using quoted symbol literals, there does not appear to be a way to enforce a consistent quote style. It seems logical that `Style/StringLiterals` should extend its enforcement to Symbol literals as well.

--------

## Expected behavior

- `Style/StringLiterals` enforced style should apply to Symbol literals, **or**
- there should be another way to enforce quotation style for quoted symbols.

## Actual behavior

- `Style/StringLiterals` enforced style does not apply to Symbol literals, **and**
- there is no other way (AFAIK) to enforce quotation style for quoted symbols.

### Setup

```yaml
# .rubocop.yml
AllCops:
  DisabledByDefault: true
Style/StringLiterals:
  Enabled: true
```

```ruby
# example.rb
puts  "abc-def" # Offensive, as expected
puts :"abc-def" # Inoffensive, but should be
```

### Output

```sh
$ rubocop example.rb
Inspecting 1 file
C

Offenses:

example.rb:2:6: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
puts "abc-def"  # Offensive
     ^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## RuboCop version

```
$ bundle exec rubocop -V
1.14.0 (using Parser 3.0.1.1, rubocop-ast 1.5.0, running on ruby 2.7.1 x86_64-darwin19)
```

**Comment by tejasbubane** [comment]
I think we should add a new boolean config option `EnforceForSymbolConversion` which if `true` will apply `EnforcedStyle` for symbol conversions. This config should be set to `false` by default to avoid breaking changes.

**Comment by bbatsov** [comment]
@tejasbubane Add this option where? I assume you're referrring to `StringLiterals`, but I don't like the idea of piling too much on individual cops. If there's a need for consistent quoting in symbol literals this should be a dedicated cop IMO.
